packages/Python/lldbsuite/test/lang/swift/implementation_only_imports/main_executable/TestMainExecutable.py

```diff
@@ -0,0 +1,86 @@
+# TestMainExecutable.py
+#
+# This source file is part of the Swift.org open source project
+#
+# Copyright (c) 2019 Apple Inc. and the Swift project authors
+# Licensed under Apache License v2.0 with Runtime Library Exception
+#
+# See https://swift.org/LICENSE.txt for license information
+# See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+#
+# ------------------------------------------------------------------------------
+"""
+Test `@_implementationOnly import` in the main executable
+"""
+import commands
+import lldb
+from lldbsuite.test.lldbtest import *
+import lldbsuite.test.decorators as decorators
+import lldbsuite.test.lldbutil as lldbutil
+import os
+import os.path
+import time
+import unittest2
+
+class TestMainExecutable(TestBase):
+
+    mydir = TestBase.compute_mydir(__file__)
+
+    def launch_and_break(self):
+        exe_name = "main"
+        exe_path = self.getBuildArtifact(exe_name)
+        target = self.dbg.CreateTarget(exe_path)
+        self.assertTrue(target, VALID_TARGET)
+        
+        source_name = "main.swift"
+        source_spec = lldb.SBFileSpec(source_name)
+        breakpoint = target.BreakpointCreateBySourceRegex("break here", source_spec)
+        self.assertTrue(breakpoint.GetNumLocations() > 0, "Breakpoint set sucessfully")
+        
+        process = target.LaunchSimple(None, None, os.getcwd())
+        self.assertTrue(process, PROCESS_IS_VALID)
+    
+    @decorators.swiftTest
+    def test_implementation_only_import_main_executable(self):
+        """Test `@_implementationOnly import` in the main executable"""
+        self.build()
+        def cleanup():
+            lldbutil.execute_command("make cleanup")
+        self.addTearDownHook(cleanup)
+        self.launch_and_break()
+
+        self.expect("fr var", substrs=[
```

Comment:
A slightly more robust way of writing this is
```
var = frame.FindVariable("value")
first = var.GetChildMemberWithName("first")
lldbutil.check_variable(self, first, value="2")
...
```

this way you don't depend on LLDB's formatting

Reply:
I was deliberately not doing that because it made it harder to observe the structure and gave worse error messages when it failed. I admit to thrashing around on this test case, though.